examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -429,3 +447,51 @@ export const TaskListInstantiationFactory = new DataObjectFactory<TaskList>(
 	[SharedCell.getFactory(), SharedString.getFactory(), SharedMap.getFactory()],
 	{},
 );
+
+export class BaseDocument extends DataObject implements IBaseDocument {
+	private readonly taskListCollection = new Map<string, TaskList>();
+
+	public readonly addTaskList = async (props: IBaseDocumentInitialState): Promise<void> => {
+		if (this.taskListCollection.has(props.externalTaskListId)) {
+			throw new Error(
+				`task list ${props.externalTaskListId} already exists on this collection`,
+			);
+		}
+		const taskList = await TaskListInstantiationFactory.createChildInstance(
+			this.context,
+			props,
+		);
+		this.taskListCollection.set(props.externalTaskListId, taskList);
+
+		// Storing the handles here are necessary for non leader
+		// clients to rehydrate local this.taskListCollection in hasInitialized().
+		this.root.set(props.externalTaskListId, taskList.handle);
+
+		console.log(this.taskListCollection.get(props.externalTaskListId));
+		this.emit("taskListCollectionChanged");
+	};
+
+	public readonly getTaskList = (id: string): TaskList | undefined => {
+		return this.taskListCollection.get(id);
+	};
+
+	protected async hasInitialized(): Promise<void> {
+		for (const [id, taskListHandle] of this.root) {
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+			const [taskListResolved] = await Promise.all([
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+				taskListHandle.get(),
+			]);
+			// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+			this.taskListCollection.set(id, taskListResolved);
+		}
```

Comment:
Was this supposed to just start the promises into the array and then do `await Promise.all()` afterwards? Right now it's going through the existing handles one by one sequentially. That said, sequentially seems fine for an example app to keep it easier to read, but in that case we can directly `await taskListHandle.get()`.